File: dotnet/src/Microsoft.Agents.Validation/Validators/ContentValidators/TextContentValidator.cs
using Microsoft.Agents.Xml.Generated.Models;

namespace Microsoft.Agents.Xml.Validation.Validators.ContentValidators;

/// <summary>
/// Validator for TextContent.
/// </summary>
public class TextContentValidator : ContentValidatorBase<TextContent>
{
    private const int MaxTextLength = 100000;

    public override ValidationResult Validate(TextContent content, ValidationContext? context = null)
    {
        var errors = new List<ValidationError>();

        // CNT-001: TextContent text must be non-empty
        var emptyError = ValidateNotEmpty(content.Text, "Text",
            ValidationErrorCode.CNT_001,
            "TextContent text must be non-empty");
        if (emptyError != null)
            errors.Add(emptyError);

        // CNT-002: TextContent text must not exceed 100,000 characters
        if (!string.IsNullOrEmpty(content.Text) && content.Text.Length > MaxTextLength)
        {
            errors.Add(CreateError(
                ValidationErrorCode.CNT_002,
                $"TextContent text must not exceed {MaxTextLength:N0} characters (actual: {content.Text.Length:N0})",
                "Text"));
        }

        return new ValidationResult(errors);
    }
}

File: typescript/packages/agents-protocol-validation/src/validators/content/TextContentValidator.ts
import type { TextContent } from '@microsoft/agents-protocol-abstractions';
import type { ValidationContext } from '../../core/ValidationContext';
import { ContentValidatorBase } from '../base/ContentValidatorBase';
import { ValidationResult } from '../../core/ValidationResult';
import { ValidationErrorCode } from '../../core/ValidationErrorCode';

/**
 * Validator for TextContent.
 */
export class TextContentValidator extends ContentValidatorBase<TextContent> {
  public validate(content: TextContent, context?: ValidationContext): ValidationResult {
    const errors = this.collectErrors(
      // CNT-001: text must be non-empty
      this.validateNotEmpty(
        content.text,
        'text',
        ValidationErrorCode.CNT_001,
        'TextContent text must be non-empty'
      ),

      // CNT-002: text must not exceed 100,000 characters
      this.validateMaxLength(
        content.text,
        100000,
        'text',
        ValidationErrorCode.CNT_002,
        'TextContent text must not exceed 100,000 characters'
      )
    );

    return new ValidationResult(errors);
  }
}
